x/oracle/keeper/oracle.go

```diff
@@ -116,6 +119,37 @@ func (k Keeper) validateApproveOracleRegistration(ctx sdk.Context, msg *types.Ms
 	return nil
 }
 
+func (k Keeper) UpdateOracleInfo(ctx sdk.Context, msg *types.MsgUpdateOracleInfo) error {
+	oracle, err := k.GetOracle(ctx, msg.OracleAddress)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrUpdateOracle, err.Error())
+	}
+
+	if oracle.OracleCommissionRate.Equal(msg.OracleCommissionRate) && oracle.Endpoint == msg.Endpoint {
+		return sdkerrors.Wrapf(types.ErrUpdateOracle, "no information to be updated")
+	}
+
+	blockTime := ctx.BlockHeader().Time
+
+	if !oracle.OracleCommissionRate.Equal(msg.OracleCommissionRate) {
+		if err := oracle.ValidateOracleCommission(blockTime, msg.OracleCommissionRate); err != nil {
+			return sdkerrors.Wrapf(types.ErrUpdateOracle, err.Error())
+		}
+		oracle.OracleCommissionRate = msg.OracleCommissionRate
+	}
+
+	if len(msg.Endpoint) > 0 {
+		oracle.Endpoint = msg.Endpoint
+	}
+
+	oracle.UpdateTime = blockTime
```

Comment:
I think the update time should be updated only when the comm rate is changed.

Reply:
That's right. It's fixed.